lib/iris/tests/stock/netcdf.py

```diff
@@ -8,12 +8,56 @@
 from pathlib import Path
 from string import Template
 import subprocess
+from typing import Optional
 
 import dask
 from dask import array as da
 import netCDF4
 import numpy as np
 
+from iris.tests import env_bin_path
+
+NCGEN_PATHSTR = str(env_bin_path("ncgen"))
+
+
+def ncgen_from_cdl(
+    cdl_str: Optional[str], cdl_path: Optional[str], nc_path: str
+):
+    """
+    Generate a test netcdf file from a cdl string.
+
+    Parameters
+    ----------
+    cdl_str : str or None
+        String containing a CDL description of a netcdf file.
+        If None, 'cdl_path' must be an existing file.
+    cdl_path : str
```

Comment:
This should have "or None"
Will fix that when I see what the tests are now doing

Reply:
Done